CR: ENGR00028658
Patch name: ENGR00028658_Adding_Clock_Enable_disable_Onewire.patch
Date: Wed Mar 14 14:16:48 2007 +0530
Source: Freescale Semiconductor, Inc.
Description: Adding clock enable & disable in Onewire for all platforms
===================================================================
diff -uNpr linux-2.6.19.2.orig/arch/arm/mach-mx3/clock.c linux-2.6.19.2/arch/arm/mach-mx3/clock.c
--- linux-2.6.19.2.orig/arch/arm/mach-mx3/clock.c	2007-04-25 17:39:26.000000000 -0600
+++ linux-2.6.19.2/arch/arm/mach-mx3/clock.c	2007-04-25 17:39:28.000000000 -0600
@@ -759,6 +759,10 @@ static struct clk i2c_clk[] = {
 static struct clk owire_clk = {
 	.name = "owire_clk",
 	.parent = &perclk_clk,
+	.enable_reg = MXC_CCM_CGR1,
+	.enable_shift = MXC_CCM_CGR1_OWIRE_OFFSET,
+	.enable = _clk_enable,
+	.disable = _clk_disable,
 };
 
 static struct clk sdhc_clk[] = {
diff -uNpr linux-2.6.19.2.orig/drivers/w1/masters/mxc_w1.c linux-2.6.19.2/drivers/w1/masters/mxc_w1.c
--- linux-2.6.19.2.orig/drivers/w1/masters/mxc_w1.c	2007-04-25 17:38:06.000000000 -0600
+++ linux-2.6.19.2/drivers/w1/masters/mxc_w1.c	2007-04-25 17:39:28.000000000 -0600
@@ -344,7 +344,6 @@ static int __devinit mxc_w1_probe(struct
 		return -ENOMEM;
 	}
 	dev->clk = clk_get(&pdev->dev, "owire_clk");
-
 	dev->bus_master = (struct w1_bus_master *)(dev + 1);
 	dev->found = 1;
 	dev->clkdiv = (clk_get_rate(dev->clk) / 1000000) - 1;
@@ -389,6 +388,7 @@ static int mxc_w1_remove(struct platform
 {
 	struct mxc_w1_device *dev = platform_get_drvdata(pdev);
 
+	clk_disable(dev->clk);
 	clk_put(dev->clk);
 	if (dev->found) {
 		w1_remove_master_device(dev->bus_master);
